src/day11.rs

```diff
@@ -1,25 +1,375 @@
 // Potential improvements:
 //
 
-pub fn day11(_input_lines: &[Vec<String>], _solution_ver: &str) -> (String, String) {
-    let answer1 = 0;
-    let answer2 = 0;
-    (format!("{}", answer1), format!("{}", answer2))
+use std::{thread, sync::{mpsc, Arc, Mutex}, fmt::Display, collections::BTreeMap};
+
+use rayon::prelude::*;
+use threadpool::ThreadPool;
+
+pub fn day11(input_lines: &[Vec<String>], _solution_ver: &str) -> (String, String) {
+    let grid_sn:i32 = input_lines[0][0].parse().unwrap();
+    let grid:Vec<Vec<i32>> = (1..301).map(|x| {
+        let row = (1..301).map(|y| power_level(x, y, grid_sn)).collect();
+        row
+    }).collect();
+
+    let answer1 = largest_3x3(&grid);
+    let answer2 = largest_nxn(&grid);
+    (format!("{},{}", answer1.0, answer1.1), format!("{},{},{}", answer2.0, answer2.1, answer2.2))
+}
+
+fn power_level(x:i32, y:i32, grid_sn:i32) -> i32{
+    // let out = if (x==90+16 || y==269+16) {true} else {false};
+    let rack_id = x+10;
+    // if out {println!("rack id:{}", rack_id);}
+    let mut pl = rack_id * y;
+    // if out {println!("pl pre sn:{}, sn:{}", pl,grid_sn);}
+    pl += grid_sn;
+    // if out {println!("pl pre mult: {}", pl);}
+    pl *= rack_id;
+    // if out {println!("pl pre 100s char:{}", pl);}
+    pl = hundreds_digit(pl);
+    // if out {println!("pl plus 5:{}", pl);}
+    pl -= 5;
+    // if out {println!("pl:{}", pl);}
+    pl
+}
+
+fn hundreds_digit(pl:i32) -> i32 {
+    pl.to_string().chars().rev().nth(2).unwrap_or_else(|| {println!("Couldn't unwrap, set to 0");'0'}).to_digit(10).unwrap() as i32
+}
+
+fn largest_3x3(grid: &Vec<Vec<i32>>) -> (i32, i32) {
+    let arc_grid = Arc::new(grid);
+    let mut max_power = Mutex::new((0, 0, 0)); // 0th is power, 1st is x, 2nd is y (should go write a struct really..)
+    rayon::scope(|s|
```

Comment:
I don't know much about Rust concurrency so my review of how you use it is going to be fairly surface level.  The docs for rayon::scope() state "**Prefer `join()` (or, even better, parallel iterators) where possible.**".  Is there a reason those alternatives don't work here?  (Probably a good topic for discussion during the meeting rather than responding to me directly).